app/models/group_assignment_invitation.rb

```diff
@@ -40,6 +43,10 @@ def assign_key
     self.key ||= SecureRandom.hex(16)
   end
 
+  def assign_short_key
+    self.short_key ||= SecureRandom.urlsafe_base64(6)
```

Comment:
Looks like `assign_short_key` could be extracted out to a model concern, would you mind doing that?